telemetry/data_window.go

```diff
@@ -80,6 +110,123 @@ var (
 	subWindowsLock    = sync.RWMutex{}
 )
 
+func getSQLSum(sqlTypeDeta *sqlType) int64 {
+	result := sqlTypeDeta.Use
+	result += sqlTypeDeta.Show
+	result += sqlTypeDeta.Begin
+	result += sqlTypeDeta.Commit
+	result += sqlTypeDeta.Rollback
+	result += sqlTypeDeta.Insert
+	result += sqlTypeDeta.Replace
+	result += sqlTypeDeta.Delete
+	result += sqlTypeDeta.Update
+	result += sqlTypeDeta.Select
+	result += sqlTypeDeta.Other
+	return result
+}
+
+func readSQLMetric(timepoint time.Time, SQLResult *sqlUsageData) error {
+	ctx := context.TODO()
+	promQL := "sum(tidb_executor_statement_total{}) by (instance,type)"
+	result, err := querySQLMetric(ctx, timepoint, promQL)
+	if err != nil {
+		if err1, ok := err.(*promv1.Error); ok {
+			return errors.Errorf("query metric error, msg: %v, detail: %v", err1.Msg, err1.Detail)
+		}
+		return errors.Errorf("query metric error: %v", err.Error())
+	}
+
+	anylisSQLUsage(result, SQLResult)
+	return nil
+}
+
+func querySQLMetric(ctx context.Context, queryTime time.Time, promQL string) (result pmodel.Value, err error) {
```

Comment:
There is a similar logic in Xukai's implementation. Maybe better to modify that logic to use this as well?

Reply:
We've checked the code and maybe it is out of the scope of this PR.

I think it's fine to keep the codes here, and improve them in another PR, here is the issue to trace it: https://github.com/pingcap/tidb/issues/25179

Reply:
Thx for the comment ! We've already modify Xukai's logic to use this as well and we will improve them in another PR, using [#25179](https://github.com/pingcap/tidb/issues/25179) to trace!